app/helpers/authorize_helper.rb

```diff
@@ -0,0 +1,32 @@
+module AuthorizeHelper
+  # Return true if user is authorized for controller/action, otherwise false
+  # +options+ : Hash containing
+  #             :controller : String or symbol for the controller, defaults to params[:controller]
+  #             :action     : String or symbol for the action
+  #             :id         : Id parameter
+  #             :auth_action: String or symbol for the action, this has higher priority that :action
+  #             :auth_object: Specific object on which we may verify particular permission
+  #             :authorizer : Specific authorizer to perform authorization on (handy to inject authorizer with base collection)
+  #             :permission : Specific permission to check authorization on (handy on custom permission names)
+  def authorized_for(options)
+    action          = options.delete(:auth_action) || options[:action]
+    object          = options.delete(:auth_object)
+    user            = User.current
+    controller      = options[:controller] || params[:controller]
+    controller_name = controller.to_s.gsub(/::/, "_").underscore
+    id              = options[:id]
+    user_id         = options[:user_id].to_param
+    permission      = options.delete(:permission) || [action, controller_name].join('_')
+
+    if object.nil?
+      user.allowed_to?({ :controller => controller_name, :action => action, :id => id, :user_id => user_id }) rescue false
+    else
+      authorizer = options.delete(:authorizer) || Authorizer.new(user)
+      authorizer.can?(permission, object) rescue false
+    end
+  end
+
+  def authorizer
+    @authorizer ||= Authorizer.new(User.current, :collection => instance_variable_get("@#{controller_name}"))
```

Comment:
Can we be sure the instance variable actually exists?
Isn't `resource_name` better then `controller_name`. See [find_common](https://github.com/theforeman/foreman/blob/develop/app/controllers/concerns/find_common.rb).

Reply:
Um, could be but it looks like neither `resource_name` nor `resource_class` works here:
```ruby
error": {"message":"undefined local variable or method `resource_name' for #<#<Class:0x0000555fb06a2740>:0x0000555fb0a21308>
Did you mean?
  @resource_scope
  @resource_class"}
```